### ISSUE TITLE
Add a step-by-step user guide on adopting Falco

### ISSUE THREAD

**Issue by aijamalnk** [falcosecurity/falco-website#1007]
/area documentation

**What would you like to be added**:
Clear instructions on adopting Falco. How it integrates with other tools. What are the dependencies? What are the expectations? 

**Why is this needed**:
To set clear expectations for user about Falco. What other tools you need to get most of Falco? Concrete step-by-step guide on:
1. Getting started - show the value of Falco (what questions it addresses/problems it solves?). Smooth installation and up and running is a prerequisite. 
2. Provide good adoption path: What do we do once we have Falco?
```[tasklist]
### Tasks
- [ ] Add getting started landing page
- [ ] Add a quickstart for Falco
- [ ] Add scenario walkthrough
- [ ] Add concepts and architecture
```

**Comment by mikegcoleman** [comment]
Over the past two days we met with @LucaGuerra and @nate-double-u to review our plan for the getting started section of the website and it looks like we have agreement, so I wanted to post an outline here. 


Getting started is intended to be a new technical user's first contact with Falco. Think of it as “prework” for users to be ready to start planning PoC or Production deployments.

Our goals are to 
* Show Falco’s value as quickly as possible
* Ensure awareness of next steps
* Provide full and complete instructions for core hands-on exercises

After getting started users should know
* What falco does
* What are its core components
* The value it brings / Key use cases
* What to do next. 

There four different page associated with the getting started journey, and they should be visited in this order (Ideally)

_Landing page_:
* Provides a quick overview of Falco’s value proposition
* Needs to describe the end-to-end workflow with Falco outputting to Sidekick 
* Does not need to go deep on anything (What not how)

_Quick Start_: 
* Designed to show the core Falco features
* Walks through to install Falco on Linux, and show some simple output
* Basically a revamp of the current "try Falco on Ubuntu" walkthrough on our current docs site

_Scenario Walk Through_:
* More detailed walk through
* Represents an end-to-end scenario that shows how Falco fits into a workflow. 
* Could be Kubernetes Audit Logs -> Falco -> Prometheus 
* Covers Plugins and Sidekick

_Concepts and Architecture_:
* Show users the core Falco components
* Educate on trade-offs and considerations for production deployments
* Could include Glossary

After progressing through these four pages, practitioners should understand what falco does, a bit about how it does it, and be ready to start planning for a POC or production deployment.

**Comment by aijamalnk** [comment]
I added the deliverables as trackable tasks. Thanks Mike!

**Comment by LucaGuerra** [comment]
This PR is related: https://github.com/falcosecurity/falco-website/pull/1072
